### ISSUE TITLE
🐞 - tourAnchor is FormArray not found when element removed and added again

### ISSUE THREAD

**Issue by tqg-aantoniou** [hakimio/ngx-ui-tour#177]
### Which ngx-ui-tour-* package is the source of the bug?

md-menu

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/angular-wau2z8?file=src%2Fapp%2Fui-tour-demo.component.ts

### Description

Create a FormArray and set the first array element a tourAnchor. Start the tour and you will see that the tour works just fine. Add a second element to the array and remove the first one. Start the tour again and the step is not found.

### Angular version

17.0.8

### Ngx UI Tour version

12.0.2

### Which browsers have you used?

- [X] Chrome
- [X] Firefox
- [X] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

**Comment by tqg-aantoniou** [comment]
Nop, that's not the case. Let me try to explain again. 

I have a component on a page (part of a form array), with a tourAnchor. The tour works just fine. 

I have a button for adding and removing such components. If i add a second one and remove the first one, I confirm that the tourAnchor of the component is correct. If i start the tour it's not finding the tourAnchor.

I also tried isAsync and it just waits for ever.

**Comment by hakimio** [comment]
Please provide a reproduction of the issue.

**Comment by tqg-aantoniou** [comment]
Here you are: 
https://stackblitz.com/edit/angular-wau2z8-mxrwfo?file=src%2Fapp%2Fui-tour-demo.component.html,src%2Fapp%2Fui-tour-demo.component.ts

Steps to reproduce:

1) Click on "Start Demo UI Tour". You should see 2 steps (one on the button and one on the control). Expected behaviour.
2) Click "Add Item". You should now have 2 controls
3) Click on "Start Demo UI Tour". You should see 2 steps (one on the button and one on the control). Expected behaviour.
4) Click on "0: Control" to remove it
5) Click on "Start Demo UI Tour". Only 1 step is shows (for the button). The step for the control is not found.

**Comment by hakimio** [comment]
After adding `trackBy` function, I can not reproduce your issue:
https://stackblitz.com/edit/angular-wau2z8-li9mvs?file=src%2Fapp%2Fui-tour-demo.component.html
